src/measure_theory/lp_space.lean

```diff
@@ -94,4 +94,68 @@ by simp [snorm, hp0_lt]
 
 end zero
 
+@[simp] lemma snorm_neg {f : α → F} : snorm (-f) p μ = snorm f p μ :=
+by simp [snorm]
+
+
+section borel_space
+variable [borel_space E]
+
+lemma mem_ℒp.neg {f : α → E} (hf : mem_ℒp f p μ) : mem_ℒp (-f) p μ :=
+⟨measurable.neg hf.1, by simp [hf.right]⟩
+
+
+variable [topological_space.second_countable_topology E]
+
+lemma mem_ℒp.add {f g : α → E} (hf : mem_ℒp f p μ) (hg : mem_ℒp g p μ) (hp1 : 1 ≤ p) :
+  mem_ℒp (f+g) p μ :=
+begin
+  have hp0_lt : 0 < p, from lt_of_lt_of_le zero_lt_one hp1,
+  have hp0 : 0 ≤ p, from le_of_lt hp0_lt,
+  split,
+  { exact measurable.add hf.1 hg.1, },
+  simp_rw [pi.add_apply, ennreal.coe_rpow_of_nonneg _ hp0],
+  -- step 1: use nnnorm_add_le
+  refine lt_of_le_of_lt _ _,
```

Comment:
This is kind of hard to read, because it creates three goals, one of which is an ennreal that you will instantiate at the next line. A more readable way is to write down `I1` for the first inequality you are proving, saying : `have I1 : ..., {and then your proof}`, then `have I2 : ...`, then have `I3 : ....`, and then concluding the proof with `exact lt_of_le_of_lt (le_trans I1 I2) I3` (or even the dot version `exact (I1.trans.I2).lt_trans I3`). Another version is to write a `calc` proof. Let me push a version with the `calc` proof to show what I mean.

Reply:
I chose the refine approach over the multiple have because I wanted to give the structure of the argument before showing the different steps. We have different opinions on what is more readable there. But in any case the calc proof you pushed is much better in that regard.
until now I only used calc in short, condensed blocks. I did not think of using it that way, to give structure to the proof. Thanks for showing me!